### ISSUE TITLE
Kafka Broker Not Available

### ISSUE THREAD

**Issue by vsnine** [F5Networks/f5-telemetry-streaming#17]
##### ISSUE TYPE

 - Bug Report

##### Telemetry Streaming BUILD/ VERSION

```
f5-telemetry-1.4.0-1.noarch.rpm
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.0.7
  Build       0.0.1
  Edition     Point Release 7
  Date        Tue May 15 15:26:30 PDT 2018
```

##### SUMMARY

Configuring a Kafka consumer as shown results in the following error message after a time. This causes restnoded to restart.

```
Sun, 30 Jun 2019 04:17:21 GMT - severe: [telemetry.Kafka] error: Broker not available
Sun, 30 Jun 2019 04:17:41 GMT - finest: [telemetry] System poller cycle started
Sun, 30 Jun 2019 04:17:41 GMT - finest: [telemetry] System poller cycle finished
Sun, 30 Jun 2019 04:17:41 GMT - finest: [telemetry] Pipeline processed data of type: systemInfo
Sun, 30 Jun 2019 04:18:18 GMT - severe: [telemetry.Kafka] error: Broker not available
Sun, 30 Jun 2019 04:18:41 GMT - finest: [telemetry] System poller cycle started
Sun, 30 Jun 2019 04:18:41 GMT - finest: [telemetry] System poller cycle finished
Sun, 30 Jun 2019 04:18:41 GMT - finest: [telemetry] Pipeline processed data of type: systemInfo
Sun, 30 Jun 2019 04:18:57 GMT - error: uncaughtException: false == true date=Sun Jun 30 2019 04:18:57 GMT+0000 (UTC), pid=7582, uid=198, gid=198, cwd=/var/service/restnoded, execPath=/usr/bin/f5-rest-node, version=v4.8.0, argv=[/usr/bin/f5-rest-node, /usr/share/rest/node/src/restnode.js, -p, 8105, --logLevel, finest, -i, /var/log/restnoded/restnoded.log, -s, none], rss=142221312, heapTotal=91953520, heapUsed=81719392, external=3965888, loadavg=[0.3466796875, 0.26904296875, 0.28662109375], uptime=390882
AssertionError: false == true
,    at KafkaClient.connect.connect.attempt.async.series.KafkaClient.connectToBrokers.KafkaClient.connectToBroker.KafkaClient.refreshBrokerMetadata.async.waterfall.KafkaClient.getListGroups.async.mapValuesLimit.KafkaClient.getDescribeGroups.async.groupByLimit.sendGroupCoordinatorRequest.async.mapValuesLimit.KafkaClient.initializeBroker.KafkaClient.deleteDisconnected (/var/config/rest/iapps/f5-telemetry/nodejs/node_modules/kafka-node/lib/kafkaClient.js:669:5)
,    at null._onTimeout (/var/config/rest/iapps/f5-telemetry/nodejs/node_modules/kafka-node/lib/kafkaClient.js:655:14)
,    at Timer.listOnTimeout (timers.js:92:15)

Sun, 30 Jun 2019 04:18:57 GMT - error: uncaughtException: false == true date=Sun Jun 30 2019 04:18:57 GMT+0000 (UTC), pid=7582, uid=198, gid=198, cwd=/var/service/restnoded, execPath=/usr/bin/f5-rest-node, version=v4.8.0, argv=[/usr/bin/f5-rest-node, /usr/share/rest/node/src/restnode.js, -p, 8105, --logLevel, finest, -i, /var/log/restnoded/restnoded.log, -s, none], rss=142221312, heapTotal=91953520, heapUsed=81519424, external=3965888, loadavg=[0.3466796875, 0.26904296875, 0.28662109375], uptime=390882
AssertionError: false == true
,    at KafkaClient.connect.connect.attempt.async.series.KafkaClient.connectToBrokers.KafkaClient.connectToBroker.KafkaClient.refreshBrokerMetadata.async.waterfall.KafkaClient.getListGroups.async.mapValuesLimit.KafkaClient.getDescribeGroups.async.groupByLimit.sendGroupCoordinatorRequest.async.mapValuesLimit.KafkaClient.initializeBroker.KafkaClient.deleteDisconnected (/var/config/rest/iapps/f5-telemetry/nodejs/node_modules/kafka-node/lib/kafkaClient.js:669:5)
,    at null._onTimeout (/var/config/rest/iapps/f5-telemetry/nodejs/node_modules/kafka-node/lib/kafkaClient.js:655:14)
,    at Timer.listOnTimeout (timers.js:92:15)
```

##### STEPS TO REPRODUCE

```
{
     "class": "Telemetry",
     "controls": {
         "class": "Controls",
         "logLevel": "debug"
     },
     "My_System": {
         "class": "Telemetry_System",
         "systemPoller": {
             "interval": 60
         }
     },
     "My_Listener": {
         "class": "Telemetry_Listener",
         "port": 6514
     },
     "AConsumer": {
         "class": "Telemetry_Consumer",
         "type": "Kafka",
         "host": "xxxxxx",
         "protocol": "binaryTcpTls",
         "allowSelfSignedCert": true,
         "port": "9094",
         "topic": "metrics.network.f5",
         "authenticationProtocol": "SASL-PLAIN",
         "username": "xxxx",
         "passphrase": {
             "cipherText": "xxxx"
         },
         "trace": "/var/tmp/ts-trace.out"
     }
 }
```

**Comment by petrov-serg** [comment]
@vsnine thank you for reporting that problem!
Internal issue AUTOTOOL-491 created.

**Comment by dstokesf5** [comment]
This was resolved as part of the 1.11.0 release.
